backend/pkg/cache/cache.go

```diff
@@ -0,0 +1,104 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+)
+
+type Cache interface {
+	Set(ctx context.Context, key string, value interface{}) error
+	SetWithTTL(ctx context.Context, key string, value interface{}, ttl time.Duration) error
+	Delete(ctx context.Context, key string) error
+	Get(ctx context.Context, key string) (interface{}, error)
+}
+
+var (
+	ErrNotFound     = errors.New("key not found")
+	cleanUpInterval = 10 * time.Second
+)
+
+type cacheValue struct {
+	value     interface{}
+	expiresAt time.Time
+}
+type cache struct {
+	store           map[string]cacheValue
+	lock            sync.RWMutex
+	cleanUpInterval time.Duration
+}
+
+func New() Cache {
```

Comment:
The helm package runs the functions like `install`, `update`, `rollback` in async and required a cache to store the results. Since we have similar in-memory store in multiple places across the project I created a separate package to avoid repeated code.

Reply:
Can you please add a commit message or comment with something like that?